fennel/datasets/datasets.py

```diff
@@ -2144,3 +2199,23 @@ def visitFirst(self, obj) -> DSSchema:
             )
         output_schema.name = f"'[Pipeline:{self.pipeline_name}]->first node'"
         return output_schema
+
+    def visitWindow(self, obj) -> DSSchema:
+        output_schema = copy.deepcopy(obj.dsschema())
+        output_schema_name = f"'[Pipeline:{self.pipeline_name}]->window node'"
+
+        if len(output_schema.keys) == 0:
+            raise ValueError(
+                f"`group_by` before `window` in {self.pipeline_name} must specify at least one key"
```

Comment:
The current error syntax is to put any user specified variable in ` <> ` so {self.pipeline_name} should come in ``. We do this so that users can quickly read the error and find what pertains to them from a huge error message

Reply:
done,